### PR TITLE
refactor: set up experiments context [DET-3255]

### PR DESCRIPTION
**Pull request by hamidzr**
## Commentary (optional)

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/react/src/contexts/AppContexts.tsx [hunk 1]
```diff
@@ -38,29 +40,33 @@ const AppContexts: React.FC = () => {
     useRestApi<Agent[]>(ioAgents, { mappers: jsonToAgents });
   const [ commandsResponse, requestCommands ] =
     useRestApi<Command[]>(ioGenericCommands, { mappers: jsonToCommands });
-  const [ experimentsResponse, requestExperiments ] =
+  const [ activeExperimentsResponse, requestExperiments2 ] =
```
Comment (hkang1) [inline webui/react/src/contexts/AppContexts.tsx:43]: nit: `requestActiveExperiments`

File: webui/react/src/pages/Dashboard.tsx [hunk 2]
```diff
@@ -56,23 +53,18 @@ const Dashboard: React.FC = () => {
...
   const activeExperiments = ActiveExperiments.useStateContext();
+  const experiments = Experiments.useStateContext();
   const commands = Commands.useStateContext();
   const notebooks = Notebooks.useStateContext();
   const shells = Shells.useStateContext();
   const tensorboards = Tensorboards.useStateContext();
-  const [ experimentsResponse, requestExperiments ] =
-    useRestApiSimple<ExperimentsParams, Experiment[]>(getExperimentSummaries, {});
+  Tensorboards.useActionContext();
+
```
Comment (hkang1) [inline webui/react/src/pages/Dashboard.tsx:62]: chore: probably accidental insertion, remove?

**Comment by hkang1** [comment]
@hamidzr  question: should `updateExperimentLocally` be updated to use contexts?

**Comment by hamidzr** [comment]
> @hamidzr question: should `updateExperimentLocally` be updated to use contexts?

there is a PR open for that https://github.com/determined-ai/determined/pull/642 is this what you were mentioning?

File: webui/react/src/contexts/AppContexts.tsx (hunk 1 above)
Comment (hamidzr) [reply webui/react/src/contexts/AppContexts.tsx:43]: : ) that's better

Comment (hamidzr) [reply webui/react/src/contexts/AppContexts.tsx:43]: that's a lot better

**Pull request merged**